Poor diagnostic for conditional expression result type error

**Description**

Here's a session in the repl that demonstrates

```
Welcome to Apple Swift version 5.8 (swiftlang-5.8.0.124.2 clang-1403.0.22.11.100).
...
  3> var a = [1]
a: [Int] = 1 value {
  [0] = 1
}
  4> a.append(true ? "yes" : "no")
expression failed to parse:
error: repl.swift:4:17: error: result values in '? :' expression have mismatching types 'String' and 'String'
a.append(true ? "yes" : "no")
                ^~~~~
```

Obviously the message is nonsense, and the problem is that the conditional expression is fine on its own but has the wrong type for its context.

See the first line of the repl session for version info

--- a/lib/Sema/CSSimplify.cpp
+++ b/lib/Sema/CSSimplify.cpp
@@ -14753,8 +14753,15 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
       // means that result would attempt a type from each side if
       // one is available and that would result in two fixes - one for
       // each mismatched branch.
-      if (branchElt->forElse())
+      if (branchElt->forElse()) {
         impact = 10;
+      } else {
+        // Also increase impact for `then` branch lower than `else` to still
+        // eliminate ambiguity, but slightly worst than the average fix to avoid
+        // so the solution which record this fix wouldn't be picked over one
+        // that has contextual mismatch fix on the result of ternary expression.
+        impact = 5;
+      }
     }
     using SingleValueStmtBranch = LocatorPathElt::SingleValueStmtBranch;
     if (auto branchElt = locator->getLastElementAs<SingleValueStmtBranch>()) {

--- a/test/expr/expressions.swift
+++ b/test/expr/expressions.swift
@@ -951,3 +951,7 @@ let _ = "foo \(42 /*
 // expected-error @-3 {{cannot find ')' to match opening '(' in string interpolation}} expected-error @-3 {{unterminated string literal}}
 // expected-error @-2 {{expected expression}}
 // expected-error @-3 {{unterminated string literal}}
+
+// https://github.com/apple/swift/issues/66192
+func I66192(_: Int) {}
+I66192(true ? "yes" : "no") // expected-error{{cannot convert value of type 'String' to expected argument type 'Int'}}

